spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1632,6 +1632,9 @@ def set_text(self, text):
         """Set the text of the editor"""
         self.setPlainText(text)
         self.set_eol_chars(text)
+
+        if isinstance(self.highlighter, sh.PygmentsSH):
+            self.highlighter.make_charlist()

```

Comment:
This is probably causing the odd segfaults shown in Travis and Azure to appear. So please use `running_under_pytest` to not apply it while running our tests.